pages/admin/lessons/[lessonSlug]/[pageName]/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { gql, useQuery } from '@apollo/client'
+import { GetAppProps, withGetApp } from '../../../../../graphql'
+import { get, sortBy, toNumber, toUpper } from 'lodash'
+import React, { useMemo, useState } from 'react'
+import AdminLessonNav from '../../../../../components/admin/lessons/AdminLessonSideNavLayout'
+import AdminLessonSideNav from '../../../../../components/admin/lessons/AdminLessonSideNav'
+import AdminModuleInputs, {
+  Props
+} from '../../../../../components/admin/lessons/AdminModuleInputs'
+import Breadcrumbs from '../../../../../components/Breadcrumbs'
+import styles from '../../../../../scss/modules.module.scss'
+import navStyles from '../../../../../scss/adminLessonNav.module.scss'
+import { AdminLayout } from '../../../../../components/admin/AdminLayout'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+
+const MAIN_PATH = '/admin/lessons'
+
+const MODULES = gql`
+  query {
+    modules {
+      id
+      name
+      content
+      lesson {
+        id
+      }
+      order
+    }
+  }
+`
+
+type Module = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+  order: number
+}
+type Modules = Module[]
+
+type ContentProps = {
+  pageName?: string | string[]
+  modules: Modules
+  lessonId: number
+  refetch: Props['refetch']
+}
+
+const extractLessonOrder = (x: string) => x.split('').slice(2).join()
+
+const Content = ({ pageName, modules, lessonId, refetch }: ContentProps) => {
+  const [selectedIndex, setSelectedIndex] = useState(-1)
+  const onAddItem = () => setSelectedIndex(-1)
+  const onSelect = (item: Omit<Module, 'order'>) => setSelectedIndex(item.id)
+
+  if (pageName !== 'modules') {
+    return (
+      <h1>
+        For now, you can only access <code>/modules</code> page
+      </h1>
+    )
+  }
+
+  return (
+    <div className={styles.container__modulesPanel}>
+      <AdminLessonSideNav
+        title="Modules"
+        items={modules}
+        onAddItem={onAddItem}
+        onSelect={onSelect}
+        selectedIndex={selectedIndex}
+      />
+      <div className={styles.container__modulesPanel__inputs}>
+        <AdminModuleInputs
+          lessonId={lessonId}
+          refetch={refetch}
+          module={modules.find(module => module.id === selectedIndex)}
+        />
+      </div>
+    </div>
+  )
+}
+
+const Lessons = ({ data }: GetAppProps) => {
+  const router = useRouter()
+  const { pageName, lessonSlug } = router.query
+
+  const { lessons } = data
+  const { data: modulesData, refetch } = useQuery<{ modules: Modules }>(MODULES)
+
+  const lesson = useMemo(() => {
+    if (lessons) {
+      const lessonFromParam = lessons.find(
+        e => toNumber(extractLessonOrder(e.slug)) === toNumber(lessonSlug)
```

Comment:
Keep it simple: `find(e => e.slug === lessonSlug)`

Reply:
`e.slug` is `js0` and `lessonSlug` is `0`. We need to extract the last value from `e.slug` and turn it into a number to be able to compare it with `lessonSlug`.

Reply:
1. Using the whole slug makes sense to me
2. Using the lessonId makes sense to me

Using a part of the slug does not make sense to me, because it seems like it will be adding an additional constraint and potential for bugs in the future: All lesson slugs **must contain a number** from now on. Like in the future if we added supplemental lesson like `Dynamic Programming` then I think this implementation will be an issue.

Reply:
Yeah, this makes sense. It'll actually be inconsistent with the way we've it in `/curriculum/jsX`

Reply:
seems like our existing implementation just checks the slug directly instead of trying to convert to number: https://github.com/garageScript/c0d3-app/blob/master/pages/curriculum/%5BlessonSlug%5D.tsx#L36